.github/workflows/docs.yml

```diff
@@ -8,61 +8,231 @@ on:
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+          
       - name: Set up JDK 17
         uses: actions/setup-java@v4
         with:
           distribution: 'corretto'
           java-version: 17
-      - name: Set up Python3
+
+      - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: '3.x'
+
+      - name: Install Python Dependencies
+        run: pip3 install nbconvert mkdocs==1.4.3 mkdocs-exclude mknotebooks mkdocs-material jupyter Pygments Markdown mike
+        
       - name: Install CN fonts
         run: sudo apt-get update && sudo apt-get install fonts-arphic-uming
-      - name: install Python Dependencies
-        run: pip3 install nbconvert mkdocs==1.4.3 mkdocs-exclude mknotebooks mkdocs-material jupyter Pygments Markdown
+
+      - name: Cache IJava Kernel
+        id: ijava-cache
+        uses: actions/cache@v4
+        with:
+          path: ~/.local/share/jupyter/kernels/java
+          key: ${{ runner.os }}-ijava-kernel
+
       - name: Install IJava kernel
+        if: steps.ijava-cache.outputs.cache-hit != 'true'
         run: |
           git clone https://github.com/frankfliu/IJava.git
           cd IJava/
           ./gradlew installKernel
-      - name: checkout repo
+
+      #Phase II: Consolidate the Source Files
+      #This phase consolidates all of the source files (i.e. Markdown files and Jupyter Notebooks) that are converted into the html files that form the documenetation website. 
+      #These files are split across the DJL, DJL-Demo, and DJL-Serving Repository. They need to be consolidated locallay and inaccordance with paths specified in the mkdocs.yml file in djl/docs.
+      - name: Create website directory
+        run: |
+          cd /home/runner/work/djl
+          mkdir site
+
+      - name: Set-up Git Config
+        run: |
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+
+      - name: Checkout DJL Repo
         uses: actions/checkout@v4
-      - name: add mybinder link
+        with: 
+          ref: ${{ github.event.inputs.is-new-version == 'true' && github.event.inputs.djl-tag || 'master' }}
+
+        #This step is necessary because the index.html file in djl/index.html overwrites the index.html file Mike create for the landing page
+        #This is problematic because this index.html corresponds to the djl website rather than the docs website.
+      - name: Delete index.html 
         run: |
-          python3 tools/scripts/add_online_runner.py
-      - name: clone demos
+          rm -f index.html    
+          git add -u 
+          git commit -m "Removed index.html" || echo "No changes to commit"
+        shell: bash
+
+      - name: Clone DJL-Serving
```

Comment:
Can we use checkout actions?

Reply:
We can, I updated the workflow to use actions/checkout.